drivers/serial/uart_bm.c

```diff
@@ -0,0 +1,1661 @@
+
+	return LL_USART_IsActiveFlag_TXE(config->usart) &&
+		LL_USART_IsEnabledIT_TC(config->usart);
+}
+
+static int uart_bm_irq_tx_complete(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	return LL_USART_IsActiveFlag_TC(config->usart);
+}
+
+static void uart_bm_irq_rx_enable(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	LL_USART_EnableIT_RXNE(config->usart);
+}
+
+static void uart_bm_irq_rx_disable(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	LL_USART_DisableIT_RXNE(config->usart);
+}
+
+static int uart_bm_irq_rx_ready(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+	/*
+	 * On stm32 F4X, F1X, and F2X, the RXNE flag is affected (cleared) by
+	 * the uart_err_check function call (on errors flags clearing)
+	 */
+	return LL_USART_IsActiveFlag_RXNE(config->usart);
+}
+
+static void uart_bm_irq_err_enable(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	/* Enable FE, ORE interruptions */
+	LL_USART_EnableIT_ERROR(config->usart);
+#if !defined(CONFIG_SOC_SERIES_STM32F0X) || defined(USART_LIN_SUPPORT)
+	/* Enable Line break detection */
+	if (IS_UART_LIN_INSTANCE(config->usart)) {
+		LL_USART_EnableIT_LBD(config->usart);
+	}
+#endif
+	/* Enable parity error interruption */
+	LL_USART_EnableIT_PE(config->usart);
+}
+
+static void uart_bm_irq_err_disable(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	/* Disable FE, ORE interruptions */
+	LL_USART_DisableIT_ERROR(config->usart);
+#if !defined(CONFIG_SOC_SERIES_STM32F0X) || defined(USART_LIN_SUPPORT)
+	/* Disable Line break detection */
+	if (IS_UART_LIN_INSTANCE(config->usart)) {
+		LL_USART_DisableIT_LBD(config->usart);
+	}
+#endif
+	/* Disable parity error interruption */
+	LL_USART_DisableIT_PE(config->usart);
+}
+
+static int uart_bm_irq_is_pending(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	return ((LL_USART_IsActiveFlag_RXNE(config->usart) &&
+		 LL_USART_IsEnabledIT_RXNE(config->usart)) ||
+		(LL_USART_IsActiveFlag_TC(config->usart) &&
+		 LL_USART_IsEnabledIT_TC(config->usart)));
+}
+
+static int uart_bm_irq_update(const struct device *dev)
+{
+	return 1;
+}
+
+static void uart_bm_irq_callback_set(const struct device *dev,
+					uart_irq_callback_user_data_t cb,
+					void *cb_data)
+{
+	struct uart_bm_data *data = dev->data;
+
+	data->user_cb = cb;
+	data->user_data = cb_data;
+}
+
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+#ifdef CONFIG_UART_ASYNC_API
+
+static inline void async_user_callback(struct uart_bm_data *data,
+				struct uart_event *event)
+{
+	if (data->async_cb)
+	{
+		data->async_cb(data->uart_dev, event, data->async_user_data);
+	}
+}
+
+static inline void async_evt_rx_rdy(struct uart_bm_data *data)
+{
+	LOG_DBG("rx_rdy: (%d %d)", data->dma_rx.offset, data->dma_rx.counter);
```

Comment:
Debug log statements can be removed from these functions: these can be registered separately in the user_callback, if desired.